ui/grid_menu.py

```diff
@@ -0,0 +1,107 @@
+from time import sleep
+
+from base_ui import BaseUIElement
+from canvas import Canvas
+
+class GridMenu(BaseUIElement):
```

Comment:
Consider defining your own `validate_contents(self, contents)`. If someone passes contents that isn't a `width`x`height` array of entries, I imagine it can break in subtle ways - better warn the developer about it first =D

Reply:
or, if you base it off `BaseListBackgroundableUIElement`, its `validate_contents` will be more than suitable